common/persistence/visibility/elasticsearch/query/converter.go

```diff
@@ -0,0 +1,406 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Copyright (c) 2017 Xargin
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package query is inspired and partially copied from by github.com/cch123/elasticsql.
+package query
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/olivere/elastic/v7"
+	"github.com/xwb1989/sqlparser"
+)
+
+type (
+	Converter struct {
+		fnInterceptor FieldNameInterceptor
+		fvInterceptor FieldValuesInterceptor
+	}
+)
+
+func NewConverter(fnInterceptor FieldNameInterceptor, fvInterceptor FieldValuesInterceptor) *Converter {
+	cvt := &Converter{
+		fnInterceptor: fnInterceptor,
+		fvInterceptor: fvInterceptor,
+	}
+
+	if cvt.fnInterceptor == nil {
+		cvt.fnInterceptor = &nopFieldNameInterceptor{}
+	}
+	if cvt.fvInterceptor == nil {
+		cvt.fvInterceptor = &nopFieldValuesInterceptor{}
+	}
+
+	return cvt
+}
+
+// ConvertWhereOrderBy transforms WHERE SQL statement to Elasticsearch query.
+// It also supports ORDER BY clause.
+func (c *Converter) ConvertWhereOrderBy(whereOrderBy string) (elastic.Query, []elastic.Sorter, error) {
+	whereOrderBy = strings.Trim(whereOrderBy, " ")
+	if whereOrderBy != "" && !strings.HasPrefix(whereOrderBy, "order by ") {
```

Comment:
make `"order by "` a const string?

Reply:
I usually use `const` for strings of they are referenced more than once and especially set in one place and then checked in another. But here it is pure user input checked only once. I believe that `const` will add more complexity.